### PR TITLE
Initial Live Bidding UI

### PR DESCRIPTION
**Pull request by orta**
screen shot 2016-04-04 at 18 39 52

Only has the Storyboard structure ATM, and introduces SwiftGen for our resources/storyboards

### CODE REVIEW

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%">1 Warning</th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:warning:</td>
      <td>Needs testing on a Phone if change is non-trivial</td>
    </tr>
    
  </tbody>
</table>
  

  


<p align="right" data-meta="generated_by_danger" data-base-commit="" data-head-commit="" >
  Generated by :no_entry_sign: <a href="https://github.com/danger/danger/">danger</a>
</p>

File: Artsy/Tooling/Generated/StoryboardConstants.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
+// Generated using SwiftGen, by O.Halligon — https://github.com/AliSoftware/SwiftGen
```
Comment (ashfurrow) [inline Artsy/Tooling/Generated/StoryboardConstants.swift]: :100:

File: Artsy/View_Controllers/Live_Auctions/LiveAuctionBidButton.swift [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import QuartzCore
+import Interstellar
+import Artsy_UIButtons
+
+class LiveAuctionBidButton : ARFlatButton {
+    let progressSignal = Signal<LiveAuctionBiddingProgressState>()
+
+    override func setup() {
+        super.setup()
+        progressSignal.map(setupWithState)
```
Comment (ashfurrow) [inline Artsy/View_Controllers/Live_Auctions/LiveAuctionBidButton.swift]: Would `.next` convey better meaning here?

Comment (orta) [inline Artsy/View_Controllers/Live_Auctions/LiveAuctionBidButton.swift]: hrm, probably, let's do it

File: Artsy_Tests/App_Tests/ARWebViewCacheHostSpec.m [hunk 3]
```diff
@@ -24,7 +24,16 @@
...
     [webView stopLoading];
 
-    expect(url.path).to.equal(@"/dev/blank");
+    // HACK!
+
+    // OK, so if you're offline, the WKWebView halts the web loading process
+    // and there's no way to get to the URL that was originally requested (via the
+    // webview) so, _this test only works when online_ - which is true for CI
+
+    // So, we only do the test if the url is not nil
```
Comment (ashfurrow) [inline Artsy_Tests/App_Tests/ARWebViewCacheHostSpec.m]: Huh!

File: Artsy_Tests/View_Controller_Tests/Live_Auction/LiveAuctionBidButtonTests.swift [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+@testable
+import Artsy
+
+class LiveAuctionBidButtonTests: QuickSpec {
+
+    override func spec() {
+
+        let examples:[LiveAuctionBiddingProgressState] = [.Idle(biddingAmount: "$45,000"), .InProgress , .NetworkFail, .Success(isMaxBidder: true), .Success(isMaxBidder: false)]
+        var i = 0
+        for state in examples {
```
Comment (ashfurrow) [inline Artsy_Tests/View_Controller_Tests/Live_Auction/LiveAuctionBidButtonTests.swift]: It's also possible to write `for (i, state) in examples.enumerate()`. Good solution to map tests to uniquely-named snapshots :+1:

**Comment by ashfurrow** [comment]
Solid PR :100:

**Comment by orta** [comment]
All amended :+1:

**Pull request merged**